### PR TITLE
Fix the pagination while listing lb services for ingress

### PR DESCRIPTION
**Pull request by prachidamle**
For https://github.com/rancher/rancher/issues/18568

- Create more than 100 ingresses using a script
- Check that 100 LB services show up on rancher UI under Kubernetes stacks
- Delete the ingresses and ensure all LB services get deleted as well

### CODE REVIEW

**Comment by prachidamle** [comment]
@alena1108 this is the fix for the pagination issue. Can you review pls.

File: provider/rancher/rancher.go [hunk 1]
```diff
@@ -249,11 +249,24 @@ func (lbp *LBProvider) GetExistingConfigNames() (map[string]bool, error) {
...
 
-	if len(lbs.Data) == 0 {
+	var allLBs []client.LoadBalancerService
+	allLBs = append(allLBs, lbs.Data...)
+
+	isPartial := lbs.Pagination.Partial
+	for isPartial {
+		newlbs, err := lbs.Next()
+		if err != nil {
+			fmt.Printf("Failed to get Next set of lb services with error: %v", err)
```
Comment (alena1108) [inline provider/rancher/rancher.go]: shall we return an error? Or if there is no next, error would be returned too? And if so, would it be possible to determine the diff between "no next" and "some other error"?

Comment (prachidamle) [reply provider/rancher/rancher.go]: When there is no next, there is no error thrown. It just sets the Pagination.Partial as false. 
We should be returning an error though when err!=nil

**Comment by alena1108** [comment]
LGTM

**Pull request merged**